.github/workflows/deploy.yml

```diff
@@ -0,0 +1,79 @@
+name: Build and publish
+on:
+  push:
+    branches: [master]
+  release:
+    types: [published]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Detect release type
+      run: |-
+        if [[ "${{github.event.ref}}" == "refs/heads/master" ]]; then
+          echo "::set-output name=type::SNAPSHOT"
+        else
+          echo "::set-output name=type::RELEASE"
+        fi
+      id: release-detection
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Import Secrets
+      uses: RichiCoder1/vault-action@v0.10.2
+      with:
+        url: ${{ secrets.STAGE_VAULT_ADDR }}
```

Comment:
I guess that this shouldn't be stage but production
I recommend using `TODO` in your code whenever you add a change that you want to roll back later

Reply:
Same applies to the next 3 lines